hash/hasher_bcrypt.go

```diff
@@ -0,0 +1,43 @@
+package hash
+
+import (
+	"context"
+	"github.com/ory/kratos/driver/config"
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/bcrypt"
+)
+
+type Bcrypt struct {
+	c BcryptConfiguration
+}
+
+type BcryptConfiguration interface {
+	config.Provider
+}
+
+func NewHasherBcrypt(c BcryptConfiguration) *Bcrypt {
+	return &Bcrypt{c: c}
+}
+
+func (h *Bcrypt) Generate(ctx context.Context, password []byte) ([]byte, error) {
+	if err := validateBcryptPasswordLength(password); err != nil {
+		return nil, err
+	}
+
+	hash, err := bcrypt.GenerateFromPassword(password, int(h.c.Config(ctx).HasherBcrypt().Cost))
+	if err != nil {
+		return nil, err
+	}
+
+	return hash, nil
+}
+
+func validateBcryptPasswordLength(password []byte) error {
+	// Bcrypt truncates the password to the first 72 bytes, following the OpenBSD implementation,
+	// so if password is longer than 72 bytes, function returns an error
+	// See https://en.wikipedia.org/wiki/Bcrypt#User_input
+	if len(password) > 72 {
+		return errors.New("passwords are limited to a maximum length of 72 characters")
```

Comment:
This error will be passed down to the UI I guess. Please add it to the messages to allow translations and mapping to other text: https://github.com/ory/kratos/blob/1940a679eb6b695e22cb939fb0d8d85cebb82a1e/text/message_registration.go

I guess you would then in the flow check for this error you return here similar to https://github.com/ory/kratos/blob/4bda1f09472bd65608cafe0d5d44287487da6f70/selfservice/flow/registration/error.go#L92

But maybe @aeneasr can give better pointers.

Reply:
Right, it would be an error like this one (actually we can just add it do that file for now): https://github.com/ory/kratos/blob/master/schema/errors.go#L59-L70